account-kit/react/src/AlchemyAccountProvider.tsx

```diff
@@ -101,6 +103,19 @@ export const AlchemyAccountProvider = (
     }
   }, [authStep]);
 
+  useEffect(() => {
+    if (signer) {
+      const cancel = signer.on("emailAuthLinkingRequired", (email) => {
+        setAuthStep({ type: "otp_verify", email });
+      });
+
+      return () => {
+        cancel();
+      };
+    }
+    return undefined;
+  }, [signer, setAuthStep]);
```

Comment:
we do something similar to this for new user signup. But I think the listener is actually added to the `Modal` itself https://github.com/alchemyplatform/aa-sdk/blob/main/account-kit/react/src/components/auth/modal.tsx#L30

Main reason was to avoid misfires or lack of fires. Also it's not enough to call `setAuthStep`, I think you also need to check if you need to open the modal.

this one might be a bit trickier... in some cases (the demo app), the dev embeds the auth card directly into their dom rather than using the modal. I imagine in that setup, we shouldn't be popping up the modal at all, but keeping the card based flow.

Reply:
something we could do is add a param for `useModal` to `uiConfig`. Should default to true. 

if it's set to false, then `onEmailAuthLinkingRequired `, we can not pop the modal, otherwise we do. For add passkey on signup, we still always pop the modal.